elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -289,4 +291,28 @@ private SQLQuery toPageTotalSQL(SQLQuery sql) {
     public static String getClassAlias(Class<?> entityClass) {
         return getTypeAlias(entityClass);
     }
+
+    /**
+     * Functional interface for describing a method to supply JpaTransaction.
+     */
+    @FunctionalInterface
+    public interface TransactionCancel {
+        public void cancel(EntityManager entityManager);
+    }
+  
+    public class FutureImplementation implements Future<QueryResult> {
+        private final TransactionCancel transactionCancel;
```

Comment:
An inner class can reference the members of the outer class if it is not static.  There is no need to:
1. Define private members
2. Have a constructor with arguments.